src/robusta/core/sinks/robusta/prometheus_health_checker.py

```diff
@@ -53,16 +54,20 @@ def prometheus_connection_checks(self, global_config: dict):
             prometheus_connection = get_prometheus_connect(prometheus_params=prometheus_params)
             check_prometheus_connection(prom=prometheus_connection, params={})
 
+            self.status.prometheus = True
+
             prometheus_flags = get_prometheus_flags(prom=prometheus_connection)
             if prometheus_flags:
                 self.status.prometheus_retention_time = prometheus_flags.get('retentionTime', "")
             else:
                 self.status.prometheus_retention_time = ""
 
-            self.status.prometheus = True
-
         except Exception as e:
-            self.status.prometheus = False
+            if isinstance(e, NoPrometheusUrlFound) \
```

Comment:
I would just remove the exception PrometheusFlagsConnectionError at the end of get_prometheus_flags since it is known acceptable behavior and maybe print to the logs instead that flags could not be queried

Reply:
The reason we have the PrometheusFlagsConnectionError is to handle and rate limit VM and prom exceptions separately.